### ISSUE TITLE
Issues with buildModelFMU function

### ISSUE THREAD

**Issue by AnHeuermann** [OpenModelica/OpenModelica#10554]
### Discussed in https://github.com/OpenModelica/OpenModelica/discussions/10544

<div type='discussions-op-text'>

<sup>Originally posted by **dietmarw** April 14, 2023</sup>
I'm staring this as a discussion since I'm not sure if this is really an issue or just me not doing it right.

In connection with testing #10485 I've tried run this testscript to generate a source-code only FMU:

```mo
loadModel(Modelica);
getErrorString();
buildModelFMU(Modelica.Blocks.Sources.IntegerConstant, version="2.0", fmuType="cs",
              fileNamePrefix="ConstInteger", platforms={}, includeResources=false);
getErrorString();
```
Which according to https://build.openmodelica.org/Documentation/OpenModelica.Scripting.buildModelFMU.html should be correct.  However this will fail, because of the `platforms={}, includeResources=false` part (either of them will break the call). Now I was confused and double checked what OMEdit calls when generating a source-code-only FMU and its call is:

`buildModelFMU(Modelica.Blocks.Sources.IntegerConstant,"2.0","cs","ConstInteger",{},false)`

That call would also fail when run from a script (the OMEdit export succeeds). It will only ever succeed if I omit the last two arguments (which then will build a binary in addition to the source).

What am I doing wrong here?</div>

**Comment by kabdelhak** [comment]
I looked into it and the main issue is that the compiler has this check:
```
  // Empty array constructors are not allowed in Modelica.
  if listEmpty(inExpl) then
    Error.addSourceMessage(Error.EMPTY_ARRAY, {}, inInfo);
    fail();
  end if;
  ```
  in Static.mo:2655. It does not seem to be allowed. I removed this and replaced it with some dummy values alongside some typing fixes, but it doesn't do the trick. It runs, but does not create a source-only FMU like OMEdit does.
  
  @perost @adeas31 Do you know if OMEdit realizes this and replaces it with another call or something like that? how would we get the same behaviour calling it from a .mos file. OMEdit reports:
  
  ```
  
[3] 16:36:22 Scripting Warning
A source-only FMU will be generated because an empty list of platforms is selected. If this is not intended, check settings in Tools->Options->FMI->Platforms.
```

**Comment by perost** [comment]
I'm not really sure what's going on with OMEdit, but `{}` is indeed forbidden in Modelica for some reason. Instead you have to use e.g. `fill("", 0)`.

**Comment by AnHeuermann** [comment]
If we can't find a way to call `buildModelFMU(myModel, platforms={})` we could add a `source-code` platform instead and in `buildModelFMU` make sure that this is the only list element. In that case one could use `source-code` or a list of other platforms, but not both at the same time, which is a bit confusing.

**Comment by dietmarw** [comment]
Having a source-code **and** other platforms at the same time is not that uncommon. That way you can generate one FMU that you can run locally on your machine for testing but still have the source-code available for some alien platform to compile on.

**Comment by casella** [comment]
Alien platform: 

![immagine](https://user-images.githubusercontent.com/4233148/232484801-f767f391-71d2-477a-bf08-e808d295b522.png)

Nice that we support that as well 😄

**Comment by AnHeuermann** [comment]
> Having a source-code and other platforms at the same time is not that uncommon.

Probably easier to implement anyway.

**Comment by dietmarw** [comment]
@casella Following the definition of Oxford Languages: 
![image](https://user-images.githubusercontent.com/9332/232531962-8be55ebc-e08a-48a9-b124-bb438029642e.png)

It pretty much describes how **Windows** appears to me :-)

**Comment by FabiJa** [comment]
@AnHeuermann Did you find a workaround? I am having the same problem. I need a [workaround ](https://github.com/OpenModelica/OpenModelica/issues/12749#issuecomment-2352372873) to programmatically generate FMUs with OM+ Windows.

**Comment by casella** [comment]
@FabiJa unfortunately @AnHeuermann left the developer's group, though he may still occasionally help with these issues. Maybe @arun3688 could take care of this?

**Comment by arun3688** [comment]
@FabiJa can you provide your use case, if i understood it correctly you want to generate fmu with only source code generated and then you want to compile manually compile the source code to generate windows binaries or whatever platform you want ?

**Comment by FabiJa** [comment]
Yes, this is my usecase as long as I cannot generate FMU binaries with OM+windows. At the moment, I am trying the current nightly build of OM according to your advice in the other issue, which seems to work.
Nevertheless, it would be nice to the option to generate source code only FMU not only from the OMEdit gui, but also using OMPython or omc.
